### DISCUSSION TITLE
VSCode integration?

### DISCUSSION THREAD

**Discussion by thomcom** [i3/i3#5500] [Q&A]
Like many other i3 users, I spend a lot of time in VSCode. I currently switch focus between VSCode sub-windows with `Ctrl-<Arrow Key>` directions, but I'd rather have the VSCode focus be integrated with i3. In i3 I hop between focused windows with `Alt-<Arrow Key>`. Does anyone have an experience with VSCode integration with i3? I'd like for it to function like another set of windows in i3 and interact with `Alt-<Arrow Key>` exactly the same way that i3 does. Is there a path forward toward this functionality?

**Comment by okraits** [comment]
How do you know that many i3 users use VS Code? :grin:

You need to check wether VS Code sub-windows can be controlled from outside. Then you could possibly write a script using the IPC interface.
